app/actions/action_pins.js

```diff
@@ -1,5 +1,37 @@
-export default function dropNewPin() {
+import { DROP_NEW_PIN } from '../constants/constants.js';
+import {store } from 'redux';
+let nextPinId = 0;
+
+
+// const fire = new Firebase("https://interruptedlobster.firebaseio.com/user1");
+
+function dropNewPin(pinInfo) {
   return {
-    type: DROP_NEW_PIN
+    type: DROP_NEW_PIN,
+    id: nextPinId++,
+    long: pinInfo.long,
+    lat: pinInfo.lat,
```

Comment:
To take that a step further, we could also storem in a child object, `coords` and just pass it directly. It'd probably also be better in general to explicitly take multiple params (rather than as `pinInfo` properties):
```
dropNewPin(coords, title) {
  return {
    type: DROP_NEW_PIN,
    coords,
    title
  };
}
```
(Where `coords` is an object with `longitude` and `latitude` properties, as from `getCurrentPosition`).